features/step_definitions/steps.rb

```diff
@@ -727,13 +727,23 @@
   expect(page).to have_text('Text component was successfully updated.')
 end
 
-Given(/^our sensor live report has a channel "([^"]*)" with the id (\d+)$/) do |name, id|
-  @channel = create(:channel, name: name, id: id)
+Given(/^our sensor live report has a channel "([^"]*)"$/) do |name|
+  @channel = create(:channel, name: name, report: Report.current)
+end
+
+Given(/^a topic "([^"]*)"$/) do |name|
+  @topic = create(:topic, name: name)
 end
 
 Given(/^we created a text component for it that is active right now$/) do
-  main_part = "If you think there is good in everybody, you haven't met everybody."
-  create(:text_component, :active, main_part: main_part, channels: [@channel])
+  create(
+    :text_component,
+    :active,
+    main_part: "Got milk?",
+    channels: [@channel],
+    topic: @topic,
+    triggers: [create(:trigger, priority: "high")]
```

Comment:
By looking at the [code on Github](https://github.com/roschaefer/story.board/blob/master/app/models/text_component.rb#L28), you can see that the priority of a `TextComponent` is the highest priority of it's `triggers`. No triggers means `select(&:priority_index)` will return an empty array.